src/Library/demos/Navigation View/code.rs

```diff
@@ -0,0 +1,38 @@
+use crate::workbench;
+use adw::prelude::*;
+
+use crate::glib::clone;
+use gtk::glib;
+
+pub fn main() {
+    let nav_view: adw::NavigationView = workbench::builder().object("nav_view").unwrap();
+    let nav_pageone: adw::NavigationPage = workbench::builder().object("nav_pageone").unwrap();
+    let next_button: gtk::Button = workbench::builder().object("next_button").unwrap();
+    let previous_button: gtk::Button = workbench::builder().object("previous_button").unwrap();
+    let nav_pagetwo: adw::NavigationPage = workbench::builder().object("nav_pagetwo").unwrap();
+    let nav_pagethree: adw::NavigationPage = workbench::builder().object("nav_pagethree").unwrap();
+    let nav_pagefour: adw::NavigationPage = workbench::builder().object("nav_pagefour").unwrap();
+    let title: gtk::Label = workbench::builder().object("title").unwrap();
+
+    next_button.connect_clicked(
+        clone!(@weak nav_view, @weak nav_pageone, @weak nav_pagefour => move |_| {
+            let current_page = nav_view.visible_page().unwrap();
+            if current_page == nav_pageone {
+                nav_view.push(&nav_pagetwo);
+            } else if current_page == nav_pagetwo {
+                nav_view.push(&nav_pagethree);
+            } else if current_page == nav_pagethree {
+                nav_view.push(&nav_pagefour);
+            }
```

Comment:
Good opportunity to use `match` :)

Reply:
It may work like this. But not sure if we can apply `match` for a `NavigationPage` variable directly.
```
        match current_page {
            x if x == nav_pageone => nav_view.push(&nav_pagetwo),
            x if x == nav_pagetwo => nav_view.push(&nav_pagethree),
            x if x == nav_pagethree => nav_view.push(&nav_pagefour),
            _ => (),
        }
```
Following gives a behavior where only first arm is executed in any condition.
```
        match current_page {
            nav_pageone => nav_view.push(&nav_pagetwo),
            nav_pagetwo => nav_view.push(&nav_pagethree),
            nav_pagethree => nav_view.push(&nav_pagefour),
            _ => (),
        }
```

Reply:
Ah yes, fair point!